libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1563,6 +1563,11 @@ void IRGeneratorForStatements::endVisit(MemberAccess const& _memberAccess)
 				"balance(" <<
 				expressionAsType(_memberAccess.expression(), *TypeProvider::address()) <<
 				")\n";
+		else if (member == "codehash")
+			define(_memberAccess) <<
+				"extcodehash(" <<
+				expressionAsType(_memberAccess.expression(), *TypeProvider::address()) <<
```

Comment:
The third parameter `forceCleanup` is default `false` for this call, whereas it is true for the old codegen. Is this asymmetry in the codegens deliberately?

Reply:
Yes, the new codegen performs way more checks.